simtools/corsika/corsika_config.py

```diff
@@ -0,0 +1,522 @@
+import logging
+import random
+
+import astropy.units as u
+from astropy.io.misc import yaml
+
+import simtools.config as cfg
+import simtools.io_handler as io
+import simtools.util.general as gen
+from simtools.util import names
+from simtools.layout.layout_array import LayoutArray
+from simtools.util.general import collectDataFromYamlOrDict
+
+__all__ = [
+    'CorsikaConfig',
+    'MissingRequiredInputInCorsikaConfigData',
+    'InvalidCorsikaInput'
+]
+
+
+class MissingRequiredInputInCorsikaConfigData(Exception):
+    pass
+
+
+class InvalidCorsikaInput(Exception):
+    pass
+
+
+class CorsikaConfig:
+    '''
+    CorsikaConfig deals with configuration for running CORSIKA. \
+    User parameters must be given by the corsikaConfigData or \
+    corsikaConfigFile arguments. An example of corsikaConfigData follows \
+    below.
+
+    .. code-block:: python
+
+    corsikaConfigData = {
+        'primary': 'proton',
+        'nshow': 10000,
+        'nrun': 1,
+        'zenith': 20 * u.deg,
+        'viewcone': 5 * u.deg,
+        'erange': [10 * u.GeV, 100 * u.TeV],
+        'eslope': -2,
+        'phi': 0 * u.deg,
+        'cscat': [10, 1500 * u.m, 0]
+    }
+
+    The remaining CORSIKA parameters can be set as a yaml file, using the argument \
+    corsikaParametersFile. When not given, corsikaParameters will be loaded \
+    from data/corsika/corsika_parameters.yml.
+
+    Attributes
+    ----------
+    site: str
+        North or South.
+    layoutName: str
+        Name of the layout.
+    layout: LayoutArray
+        Instance of LayoutArray.
```

Comment:
'A LayoutArray is an Instance of a LayoutArray'...maybe find a better description? :-)

Reply:
I replaced it by a more detailed description: 'LayoutArray object containing the telescope positions.'

These docstrings are redundant many times, but that is inevitable. Some times the variable is perfectly described by the name but we still need to add a description.

Reply:
Yes, this sounds much better. Sorry to be so pedantic with my comments - but obviously one should learn something from reading a description. Thanks!